src/cmds/qdel.c

```diff
@@ -241,25 +328,23 @@ char **envp;
 			pbs_strncpy(warg, warg1, sizeof(warg));
 		}
 
-		stat = pbs_deljob(connect, job_id_out, warg);
+		p_delstatus = pbs_deljoblist(connect, pbs_svr->jobids, warg);
+		
+		while (p_delstatus != NULL) {
+			if ((pbse_to_txt(p_delstatus->code) != NULL) && (p_delstatus->code != PBSE_HISTJOBDELETED)) {
+				fprintf(stderr, "%s: %s %s\n", "qdel", pbse_to_txt(p_delstatus->code), p_delstatus->name);
+				any_failed = p_delstatus->code;
+			}
 
-		/*
-		 * The counter num_deleted should not be updated  when a history job is deleted .
-		 */
-		if (pbs_errno != PBSE_HISTJOBDELETED)
-			num_deleted++;
-		if (stat && (pbs_errno != PBSE_UNKJOBID && pbs_errno != PBSE_HISTJOBDELETED)) {
-			prt_job_err("qdel", connect, job_id_out);
-			any_failed = pbs_errno;
-		} else if (stat && (pbs_errno == PBSE_UNKJOBID) && !located) {
-			located = TRUE;
-			if (locate_job(job_id_out, server_out, rmt_server)) {
-				pbs_disconnect(connect);
-				pbs_strncpy(server_out, rmt_server, sizeof(server_out));
-				goto cnt;
+			if (p_delstatus->code != PBSE_HISTJOBDELETED)
+				num_deleted++;
```

Comment:
So, if the code is 'PBSE_UNKJOBID', you will still increment num_deleted here, which seems incorrect. Maybe check for code to be PBSE_NONE to increment num_deleted?